src/Microsoft.Health.Dicom.Core/Features/CustomTag/ICustomTagStore.cs

```diff
@@ -39,11 +39,12 @@ public interface ICustomTagStore
         Task<IEnumerable<CustomTagEntry>> GetAllCustomTagsAsync(CancellationToken cancellationToken = default);
 
         /// <summary>
-        /// Delete custom tag from CustomTagStore.
+        /// Delete custom tag.
         /// </summary>
-        /// <param name="key">The tag key.</param>
-        /// <param name="cancellationToken">The cancellation token.</param>
-        /// <returns>Result.</returns>
-        Task DeleteCustomTagAsync(long key, CancellationToken cancellationToken = default);
+        /// <param name="tagPath">The tag path.</param>
+        /// <param name="customTagDataType">The custom tag data type.</param>
+        /// <param name="cancellationToken">The cancellation token</param>
+        /// <returns>The task.</returns>
+        Task DeleteCustomTagAsync(string tagPath, CustomTagDataType customTagDataType, CancellationToken cancellationToken = default);
```

Comment:
I think the `CustomTagDataType` enum sort of implies that there should be separate handling per VR, but that seems more like an implementation detail. I would say to either:
1. Pass `CustomTagEntry` to the method instead of both `tagpath` and `customTagDataType` OR
2. Replace `customTagDataType` with the VR directly

Nevertheless, I think the concrete implementation should perform some logic equivalent to the current `CustomTagLimit` class by analyzing the VR.